### ISSUE TITLE
Allowed memory size

### ISSUE THREAD

**Issue by wehowski** [friendica/friendica#14061]
Since ~2 days I have the error from the worker

PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 87721956 bytes) in .../src/Object/Image.php on line 231

VersionFriendica 'Yellow Archangel' 2024.03 - 1557
Servereinstellungen
PHP
version	8.3.4
php.ini	/opt/plesk/php/8.3/etc/php.ini
upload_max_filesize	2M
post_max_size	8M
memory_limit	512M
MySQL / MariaDB
max_allowed_packet	16777216

Could it help to use the addon "leistungsschutzrecht"? I guess it is about the post of some media (in)?

**Comment by MrPetovan** [comment]
> Could it help to use the addon "leistungsschutzrecht"? I guess it is about the post of some media (in)?

Unfortunately it wouldn't. This addon does a very specific image excision (mostly in link previews) and it doesn't include posts received through the regular ActivityPub protocol.

However, I note a discrepancy between the limit that PHP hit (134217728 bytes ~= 128M) and the limit you reported set in the php.ini (512M).

Please make sure the limit is actually set by reloading either your web server or your phpfpm, and check in the Admin -> Summary page that it is correctly set.

**Comment by wehowski** [comment]
Ahje, thank you very much that helped me a lot!
The webaccounts setting was not accepted, so I needed to edit the server wide php.ini and was now able to run the worker without error.

**Comment by wehowski** [comment]
I have "buffer" issues now in the cronjob:

PHP Fatal error:  gd-webp cannot allocate temporary buffer in /.../src/Object/Image.php on line 231

I have memory_limit set to 1024M.
I could try to increase it in the PHP.ini more? But I am not quite sure if I have to move server for better performance?

Is there a requirement published anywhere for memor_limit and other settings? (Sorry for asking again)

**Comment by MrPetovan** [comment]
1024M should be enough. `gd-webp` is an extension that handles animation files in WEBP format, so these can be quite large depending on their size and length. I would like us to gracefully handle these files by not generating preview for them, but I'm not sure how to go about it yet.

**Comment by wehowski** [comment]
Thank you for helping @MrPetovan !
- I tested around the friendica (worker) settings but without any success.
- I then disabled gd extension server wide and could run the worker without the error, BUT the gd extension was needed by another application on the server and broke
- My Solution: I disbled gd extension just for an extra Instance of PHP 8.2 I run the workers in, while friendica web is still running on the PHP 8.3 instance.

Thank you!
